implementations/micrometer-registry-dynatrace/src/main/java/io/micrometer/dynatrace/v2/DynatraceExporterV2.java

```diff
@@ -0,0 +1,362 @@
+ */
+
+package io.micrometer.dynatrace.v2;
+
+import com.dynatrace.metric.util.*;
+import io.micrometer.core.instrument.Timer;
+import io.micrometer.core.instrument.*;
+import io.micrometer.core.instrument.distribution.HistogramSnapshot;
+import io.micrometer.core.instrument.distribution.ValueAtPercentile;
+import io.micrometer.core.instrument.util.AbstractPartition;
+import io.micrometer.core.ipc.http.HttpSender;
+import io.micrometer.dynatrace.AbstractDynatraceExporter;
+import io.micrometer.dynatrace.DynatraceConfig;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.net.MalformedURLException;
+import java.net.URL;
+import java.time.Instant;
+import java.util.*;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+import java.util.stream.StreamSupport;
+
+/**
+ * @author Georg Pirklbauer
+ */
+public class DynatraceExporterV2 extends AbstractDynatraceExporter {
+    private final String endpoint;
+    private final MetricBuilderFactory metricBuilderFactory;
+    private final Logger logger = LoggerFactory.getLogger(DynatraceExporterV2.class.getName());
+    private static final Map<String, String> staticDimensions = new HashMap<String, String>() {{
+        put("dt.metrics.source", "micrometer");
+    }};
+
+    private static final int METRIC_LINE_MAX_LENGTH = 2000;
+
+    public DynatraceExporterV2(DynatraceConfig config, Clock clock, HttpSender httpClient) {
+        super(config, clock, httpClient);
+
+        this.endpoint = prepareEndpoint(config);
+        logger.debug(String.format("exporting to endpoint %s", endpoint));
+
+        MetricBuilderFactory.MetricBuilderFactoryBuilder factoryBuilder = MetricBuilderFactory
+                .builder()
+                .withPrefix(config.metricKeyPrefix());
+
+        if (config.defaultDimensions() != null) {
+            factoryBuilder.withDefaultDimensions(parseDefaultDimensions(config.defaultDimensions()));
+        }
+
+        if (config.enrichWithOneAgentMetadata()) {
+            factoryBuilder.withOneAgentMetadata();
+        }
+
+        metricBuilderFactory = factoryBuilder.build();
+    }
+
+    private String prepareEndpoint(DynatraceConfig config) {
+        String endpoint;
+
+        if (config.uri().matches("/metrics/ingest/?$")) {
+            endpoint = config.uri();
+        } else {
+            try {
+                endpoint = new URL(new URL(config.uri()), "/api/v2/metrics/ingest").toString();
+            } catch (MalformedURLException e) {
+                logger.warn("could not parse endpoint url. Falling back to local OneAgent endpoint.");
+                endpoint = "http://127.0.0.1:14499/metrics/ingest";
+            }
+        }
+        return endpoint;
+    }
+
+    private DimensionList parseDefaultDimensions(Map<String, String> defaultDimensions) {
+        if (defaultDimensions == null || defaultDimensions.isEmpty()) {
+            return DimensionList.create(Dimension.create("dt.metrics.source", "micrometer"));
```

Comment:
This duplicates the definition of the dimension so one might overlook one of them when something is to be changed here. Please avoid this (e.g., by setting defaultDimensions to an empty list so the code below takes care of it).